models/issue_xref.go

```diff
@@ -0,0 +1,309 @@
+				}
+				return nil, err
+			}
+			if err = ctx.OrigIssue.loadRepo(e); err != nil {
+				return nil, err
+			}
+			if xref, err = issue.isValidCommentReference(e, ctx, repo, index); err != nil {
+				return nil, err
+			}
+			if xref != nil {
+				xreflist = issue.updateCrossReferenceList(xreflist, xref)
+			}
+		}
+	}
+
+	return xreflist, nil
+}
+
+func (issue *Issue) updateCrossReferenceList(list []*crossReference, xref *crossReference) []*crossReference {
+	if xref.Issue.ID == issue.ID {
+		return list
+	}
+	for i, r := range list {
+		if r.Issue.ID == xref.Issue.ID {
+			if xref.Action != XRefActionNone {
+				list[i].Action = xref.Action
+			}
+			return list
+		}
+	}
+	return append(list, xref)
+}
+
+func (issue *Issue) isValidCommentReference(e Engine, ctx *crossReferencesContext, repo *Repository, index int64) (*crossReference, error) {
+	refIssue := &Issue{RepoID: repo.ID, Index: index}
+	if has, _ := e.Get(refIssue); !has {
+		return nil, nil
+	}
+	if err := refIssue.loadRepo(e); err != nil {
+		return nil, err
+	}
+	// Check user permissions
+	if refIssue.Repo.ID != ctx.OrigIssue.Repo.ID {
+		perm, err := getUserRepoPermission(e, refIssue.Repo, ctx.Doer)
+		if err != nil {
+			return nil, err
+		}
+		if !perm.CanReadIssuesOrPulls(refIssue.IsPull) {
+			return nil, nil
+		}
+	}
+	return &crossReference{
+		Issue:  refIssue,
+		Action: XRefActionNone,
+	}, nil
+}
+
+func (issue *Issue) neuterCrossReferences(e Engine) error {
+	return neuterCrossReferences(e, issue.ID, 0)
+}
+
+// _________                                       __
+// \_   ___ \  ____   _____   _____   ____   _____/  |_
+// /    \  \/ /  _ \ /     \ /     \_/ __ \ /    \   __\
+// \     \___(  <_> )  Y Y  \  Y Y  \  ___/|   |  \  |
+//  \______  /\____/|__|_|  /__|_|  /\___  >___|  /__|
+//         \/             \/      \/     \/     \/
+//
+
+func (comment *Comment) addCrossReferences(e *xorm.Session, doer *User) error {
+	if comment.Type != CommentTypeCode && comment.Type != CommentTypeComment {
+		return nil
+	}
+	if err := comment.loadIssue(e); err != nil {
+		return err
+	}
+	ctx := &crossReferencesContext{
+		Type:        CommentTypeCommentRef,
+		Doer:        doer,
+		OrigIssue:   comment.Issue,
+		OrigComment: comment,
+	}
+	return comment.Issue.findCrossReferences(e, ctx, comment.Content)
+}
+
+func (comment *Comment) neuterCrossReferences(e Engine) error {
+	return neuterCrossReferences(e, 0, comment.ID)
+}
+
+// LoadRefComment loads comment that created this reference from database
+func (comment *Comment) LoadRefComment() (err error) {
+	if comment.RefComment != nil {
+		return nil
+	}
+	comment.RefComment, err = GetCommentByID(comment.RefCommentID)
+	return
+}
+
+// LoadRefIssue loads comment that created this reference from database
+func (comment *Comment) LoadRefIssue() (err error) {
+	if comment.RefIssue != nil {
+		return nil
+	}
+	comment.RefIssue, err = GetIssueByID(comment.RefIssueID)
+	if err == nil {
+		err = comment.RefIssue.loadRepo(x)
```

Comment:
Why we need that?

Reply:
```
err = comment.RefIssue.loadRepo(x)
```
It's used to display the repository name in the template.

Reply:
So this may be invoked when it will be used.